### ISSUE TITLE
syslog errors when starting pihole-FTL

### ISSUE THREAD

**Issue by jpgpi250** [pi-hole/pi-hole#4177]
### Versions
  Pi-hole version is v5.3.1 (Latest: v5.3.1)
  AdminLTE version is v5.5 (Latest: v5.5)
  FTL version is v5.8.1 (Latest: v5.8.1)

### Platform
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

### Expected behavior

No syslog errors when stop / start pihole-FTL

### Actual behavior / bug

when pihole-FTL starts, the following errors are logged in syslog
`
systemd[17378]: gpgconf: running /usr/bin/gpg-agent failed (exitcode=2): General error
systemd[17378]: gpgconf: fatal error (exit status 1)
`

### Steps to reproduce

- stop pihole-FTL (sudo service pihole-FTL stop)
- wait 30 seconds
- start pihole-FTL (sudo service pihole-FTL start)

**Comment by DL6ER** [comment]
> duckduckgo is your friend

No need to go this far: [Pi-hole Dirscourse :: Gpgconf error after install Pi-hole](https://discourse.pi-hole.net/t/gpgconf-error-after-install-pi-hole-gpgconf-fehler-nach-pi-hole-installation/21624).

This summarizes the issue pretty good. And clarifies it is a gnupg bug and why it should **not** be "fixed" the way you found.

TL;DR: `Bucking_hoen` said [here](https://discourse.pi-hole.net/t/gpgconf-error-after-install-pi-hole-gpgconf-fehler-nach-pi-hole-installation/21624/19):

> The problem is triggered by a systemd system-wide environment-generator for Debian's *`gpg-agent`* that gets called on user session creation (`/usr/lib/systemd/user-environment-generators/90gpg-agent`).
> [...]
> Unfortunately, **neither fixed version seems to be available for mainline Debian Buster**, though at least a backport for *`gpg-agent`* exists.
---

> I assume the user is created by the installer script.

Yes,
https://github.com/pi-hole/pi-hole/blob/4736e03108763cc2d5659f48d8a1e8a64d9b2608/automated%20install/basic-install.sh#L1838-L1888
more specifically in this line:
https://github.com/pi-hole/pi-hole/blob/4736e03108763cc2d5659f48d8a1e8a64d9b2608/automated%20install/basic-install.sh#L1881

> the user pihole is created with home directory /home/pihole, but that folder doesn't exist.

No, the user is *supposed to* have no home directory. Yes, the directory does not exist.

The option `-r` creates a system account. A system account is a user *only* used by the system without a home directory. This is correct and the reason for the "issue" here - we don't want the user to have a home directory because we don't want that users can log in as this user at all. I still believe `gnupg` is causing the issue here. Because it should not complain about a directory missing when it is perfectly fine that it is missing (system account). **What we really want** is that `gpgconf` does not try to launch an agent for a system account.

I do not see this error, even when
```
$ ls /home/
me
```
clearly indicates that there is no `pihole` user home on my system, yet, I don't see the issue you are reporting myself. But then I don't have a running Raspbian at the moment.

What are the version of your `gpg` and `gpg-agent` and `gpgconf` ?
```
$ gpg --version
gpg (GnuPG) 2.2.19
libgcrypt 1.8.5

$ gpg-agent --version
gpg-agent (GnuPG) 2.2.19

$ gpgconf --version
gpgconf (GnuPG) 2.2.19
```

**Comment by DL6ER** [comment]
Since there is even more discussion already in the Discourse topic and since there is already an answer over there, I'll close this ticket. It is a Buster issue. A fixed version exists but hasn't been backported. There are workarounds floating around on the web (create a home directory for the `pihole` user), but it should not be done because it is not intended for system accounts to have home directories at all.
